### ISSUE TITLE
Add Friend sometimes allows you to continue adding the same person

### ISSUE THREAD

**Issue by supermandrew** [Segs/Segs#515]
### Expected behavior
When right-clicking on a player, clicking Add Friend should add a player as a friend.  Right-clicking again on the same player should now show Remove Friend where Add Friend once was, thus preventing you from adding the same friend multiple times.

### Actual behavior/symptoms
When right-clicking on a player, clicking Add Friend adds a player as a friend.  Then *sometimes* (always for the duration of being logged in) right-clicking again on the same player continues to show Add Friend, allowing you to add the same player over and over again.

It's important to note that if you click any of the added friends in your friends list, while it normally will highlight the entity of your friend if they are nearby, in this case it always says not in range or offline.

Logging on and off keeps these players in your friends list, it's not just a session glitch.

### Steps to reproduce behavior/symptoms
Somewhat tough to reproduce, as it appears random.  The way I've done it is:
- Wipe database
- Create character for player A.  Log on.
- Create character on another user, player B.  Log on.
- Use player A and right-click on player B, then click Add Friend.  After that, try right-clicking on player B again - if you see Add Friend still, you're experiencing the bug.  
- If not, have player B log out, and create a new character for player B and log in once again.  Repeat the previous step again.

### SEGS version
0.5.0

### Operating system
Windows

### Additional info
Screenshots attached showing bug.
![friendglitch1](https://user-images.githubusercontent.com/19180694/43823421-f6c2ca8a-9abc-11e8-9e1e-535e3e23e506.PNG)
![friendglitch2](https://user-images.githubusercontent.com/19180694/43823422-f6ccbed2-9abc-11e8-8ab5-132ed1435a33.PNG)

**Comment by supermandrew** [comment]
I haven't been able to reproduce this without creating a new character.  For instance, if player A logs on, and player B has logged on/off before at some point, if player B logs on this bug doesn't seem to happen.

Even if it's player B's first time logging on and the bug doesn't happen for whatever reason, if player A adds B then and then B logs off, the bug doesn't ever seem to appear when B logs back on.

That might need more thorough testing though - just something I've noticed.

**Comment by deciton666** [comment]
Even when it cannot be reproduced in the right click menu, you can still add the friend multiple times using the "/friend <name>" slash command.

**Comment by broxen** [comment]
it's possible that this relates to where we send m_map_idx in FriendsListUpdate.h and Character.cpp

https://github.com/Segs/Segs/blob/a5253fc54323a7f11cabc4aa36a4c829a433f1cf/Projects/CoX/Servers/MapServer/Events/FriendsListUpdate.h#L47

That value is a guess, because our information from the client side is vague. My suspicion is that the client expects a unique ID for the map instance you're on and not the map_idx.

Client side for that line looks like this:
```
if ( online_status )
            {
                int mapidx_maybe = my_pakGetPackedBits(pak, 1);
                if ( ent && !skip_update )
                    ent->friend_list->entries[i].friend_mapidx_maybe = mapidx_maybe;
                QString mapname = my_pakGetString(pak);
// ...
```

**Comment by deciton666** [comment]
I remember from back in the days that it was very common to have multiple instances of popular maps, like Atlas Park. Each of these probably have a unique id of some sort.

**Comment by broxen** [comment]
That's true, AP had 33 instances on Virtue at shutdown. It's something we may want to consider when working on map transfers.

For this issue I'm not sure if the client is truly looking for map_idx there.

One way to test my theory (and possibly resolve this bug) would be to change that for an entity's m_idx and see what happens
